project-backend/Controllers/JobController.cs

```diff
@@ -92,6 +95,39 @@ public AddJobResponseObject AddJob([FromBody] AddJobQueryObject job)
             };
             return response;
         }
+
+        [HttpGet]
+        [Route("Details")]
+        [ProducesResponseType(typeof(JobResponseObject), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(Error), StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(typeof(ValidationProblemDetails), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(Error), StatusCodes.Status404NotFound)]
+        public IActionResult GetJobDetails([FromQuery] GetJobDetailsQueryObject job)
+        {
+            if (!int.TryParse(User.GetUserIdClaim().Value, out var userId))
+            {
+                return Unauthorized(new Error("Not logged in"));
+            }
+            try
+            {
+                var requiredJob = _jobProvider.QueryJobs().Where(j => j.Id == job.Id).First();
```

Comment:
I think this logic should go in the provider. (GetJobById)

Reply:
Totally agree!